File: web/src/main/kotlin/no/ssb/kostra/web/viewmodel/FileReportEntryVm.kt
package no.ssb.kostra.web.viewmodel

import com.fasterxml.jackson.annotation.JsonInclude
import io.micronaut.serde.annotation.Serdeable
import no.ssb.kostra.validation.report.Severity

@Serdeable
@JsonInclude(JsonInclude.Include.ALWAYS)
data class FileReportEntryVm(
    val severity: Severity,
    val caseworker: String,
    val journalId: String,
    val individId: String,
    val contextId: String,
    val ruleName: String,
    val messageText: String,
    val lineNumbers: List<Int>
)

File: web/src/main/vite-project/src/kostratypes/fileReportEntryVm.ts
import KostraSeverity from "./kostraSeverity"
import Nullable from "./nullable";

interface FileReportEntryVm {
    readonly severity: NonNullable<KostraSeverity>
    readonly caseworker: Nullable<string>
    readonly journalId: Nullable<string>
    readonly individId: Nullable<string>
    readonly contextId: Nullable<string>
    readonly ruleName: NonNullable<string>
    readonly messageText: NonNullable<string>
    readonly lineNumbers: Nullable<number[]>
}

export default FileReportEntryVm